### PR TITLE
Fix unmarshalling of additional properties

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Generator now generates UnmarshalJSON() method for types that contain
additional properties.
Add missing tests for additional properties.

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Fixes https://github.com/Azure/azure-sdk-for-go/issues/2695

File: .gulp/regeneration.iced [hunk 1]
```diff
@@ -54,6 +54,7 @@ regenExpected = (opts,done) ->
         return done()
 
 goMappings = {
+  'additional-properties':['additionalProperties.json', 'additionalproperties'],
```
Comment (vladbarosan) [inline .gulp/regeneration.iced]: is there anything other tests we are missing, if so maybe open an issue for it

File: src/Templates/ModelTemplate.cshtml [hunk 2]
```diff
@@ -223,7 +223,7 @@ else
     </text>
 }
 
-@if (Model.HasPolymorphicFields || Model.HasFlattenedFields)
+@if (Model.HasPolymorphicFields || Model.HasFlattenedFields || Model.AdditionalPropertiesField != null)
```
Comment (vladbarosan) [inline src/Templates/ModelTemplate.cshtml]: no regressions because of this ? also nit maybe extract in a needsCustomMarshalling method

Comment (jhendrixMSFT) [reply src/Templates/ModelTemplate.cshtml]: I still have to test that part, I will report back when I have those results.

File: .gulp/regeneration.iced (hunk 1 above)
Comment (jhendrixMSFT) [reply .gulp/regeneration.iced]: Right now we have a generic issue tracking this, https://github.com/Azure/azure-sdk-for-go/issues/1428

Comment (vladbarosan) [reply .gulp/regeneration.iced]: eh ok, but we dont get a test coverage report anymore right , and sorry for my grammar, sleepy

File: src/Templates/ModelTemplate.cshtml (hunk 2 above)
Comment (vladbarosan) [reply src/Templates/ModelTemplate.cshtml]: great, well approved if everything is fine feel free to merge

File: .gulp/regeneration.iced (hunk 1 above)
Comment (jhendrixMSFT) [reply .gulp/regeneration.iced]: You can find the coverage report here http://azure.github.io/autorest/dashboard.html

Comment (vladbarosan) [reply .gulp/regeneration.iced]: oh didnt know about it. can we make version 3 not show on top ? :D

**Pull request merged**